### PR TITLE
Convert all network protos to Prost

### PR DESCRIPTION
**Pull request by bmwill**
This completes converting all network protos to Prost

### CODE REVIEW

File: consensus/src/chained_bft/consensus_types/block.rs [hunk 1]
```diff
@@ -513,61 +511,31 @@ where
...
-        let id = HashValue::from_slice(object.get_id())?;
-        let parent_id = HashValue::from_slice(object.get_parent_id())?;
-        let timestamp_usecs = object.get_timestamp_usecs();
-        let round = object.get_round();
-        let height = object.get_height();
-        let quorum_cert = QuorumCert::from_proto(object.take_quorum_cert())?;
+    type Error = failure::Error;
+
+    fn try_from(proto: network::proto::Block) -> failure::Result<Self> {
+        let id = HashValue::from_slice(proto.id.as_ref())?;
```
Comment (mimoo) [inline consensus/src/chained_bft/consensus_types/block.rs]: btw, code looks so much clearer without these getters :P

File: consensus/Cargo.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ num-derive = { version = "0.2.5", default-features = false }
 num-traits = { version = "0.2.8", default-features = false }
 parity-multiaddr = { version = "0.5.0", default-features = false }
 protobuf = "~2.7"
```
Comment (mimoo) [inline consensus/Cargo.toml]: do we still need this?

File: consensus/src/chained_bft/network.rs [hunk 3]
```diff
@@ -185,19 +188,20 @@ impl ConsensusNetworkImpl {
         timeout: Duration,
     ) -> failure::Result<BlockRetrievalResponse<T>> {
         ensure!(from != self.author, "Retrieve block from self");
-        let mut req_msg = RequestBlock::new();
-        req_msg.set_block_id(block_id.into());
-        req_msg.set_num_blocks(num_blocks);
+        let mut req_msg = RequestBlock::default();
+        req_msg.block_id = block_id.to_vec();
+        req_msg.num_blocks = num_blocks;
```
Comment (mimoo) [inline consensus/src/chained_bft/network.rs]: much cleaner without setters too!

File: network/src/validator_network/consensus.rs [hunk 4]
```diff
@@ -6,23 +6,23 @@
...
-use bytes::Bytes;
 use channel;
 use futures::{
     stream::Map,
     task::{Context, Poll},
     SinkExt, Stream, StreamExt,
 };
 use pin_utils::unsafe_pinned;
-use protobuf::Message as proto_msg;
+use prost::Message as _;
```
Comment (mimoo) [inline network/src/validator_network/consensus.rs]: 👍

File: consensus/Cargo.toml (hunk 2 above)
Comment (bmwill) [reply consensus/Cargo.toml]: Yes, i have another PR that will remove this :)

**Pull request closed without merging**